src/conn/conn_evict.c

```diff
@@ -0,0 +1,356 @@
+/*-
+ * Copyright (c) 2014-present MongoDB, Inc.
+ * Copyright (c) 2008-2014 WiredTiger, Inc.
+ *	All rights reserved.
+ *
+ * See the file LICENSE for redistribution information.
+ */
+
+#include "wt_internal.h"
+
+#define WT_CONFIG_DEBUG(session, fmt, ...)                                 \
+    if (FLD_ISSET(S2C(session)->debug_flags, WT_CONN_DEBUG_CONFIGURATION)) \
+        __wt_verbose_warning(session, WT_VERB_CONFIGURATION, fmt, __VA_ARGS__);
+
+/*
+ * __evict_config_abs_to_pct --
+ *     Evict configuration values can be either a percentage or an absolute size, this function
+ *     converts an absolute size to a percentage.
+ */
+static WT_INLINE int
+__evict_config_abs_to_pct(
+  WT_SESSION_IMPL *session, double *param, const char *param_name, bool shared)
+{
+    WT_CONNECTION_IMPL *conn;
+    double input;
+
+    conn = S2C(session);
+
+    WT_ASSERT(session, param != NULL);
+    input = *param;
+
+    /*
+     * Anything above 100 is an absolute value; convert it to percentage.
+     */
+    if (input > 100.0) {
+        /*
+         * In a shared cache configuration the cache size changes regularly. Therefore, we require a
+         * percentage setting and do not allow an absolute size setting.
+         */
+        if (shared)
+            WT_RET_MSG(session, EINVAL,
+              "Shared cache configuration requires a percentage value for %s", param_name);
+        /* An absolute value can't exceed the cache size. */
+        if (input > conn->cache_size)
+            WT_RET_MSG(session, EINVAL, "%s should not exceed cache size", param_name);
+
+        *param = (input * 100.0) / (conn->cache_size);
+    }
+
+    return (0);
+}
+
+/*
+ * __evict_validate_config --
+ *     Validate trigger and target values of given configs.
+ */
+static int
```

Comment:
In future can we separate the following steps into distinct commits? It makes it easier to understand the changes when walking the commits one by one
- Moving a function to a new file, 
- Renaming functions, 
- Removing functions
- Changing a function body

A couple of examples where this would help:
- `__evict_validate_config` used to be called `__validate_cache_config`, but this is hard to determine as the diff shows `__validate_cache_config` deleted from `conn_cache.c` and a new `__evict_validate_config` function appearing in conn_evict.c
- Similarly there's changes to the function body of `__evict_validate_config` that are hidden by the move to a new file
- We discussed how `__cache_config_local` was deleted and its contents moved into `__wti_cache_config` because once we'd removed the eviction logic the function was too small to warrant a function. This isn't obvious from the [current commit](https://github.com/wiredtiger/wiredtiger/pull/11040/commits/dfb4d3386e07249b2e10c1f1318d73eb120524cd) 

CC @bvpvamsikrishna as this seems relevant to the modularity playbook and handling risk.

Reply:
Yes, that is definitely more readable and easier on reviewers, will keep in mind. :)